subsys/mpsl/pm/mpsl_pm_utils.c

```diff
@@ -71,22 +70,21 @@ void mpsl_pm_utils_work_handler(void)
 	}
 	case MPSL_PM_EVENT_STATE_BEFORE_EVENT:
 	{
-		/* Event scheduled */
-		uint64_t event_time_us = params.event_time_rel_us -
-						MAX_DELAY_SINCE_READING_PARAMS_US;
-
 		/* In case we missed a state and are in zero-latency, set low-latency.*/
 		m_update_latency_request(PM_MAX_LATENCY_HCI_COMMANDS_US);
 
-		if (event_time_us > UINT32_MAX) {
+		/* Event scheduled */
+		if (params.event_time_abs_us > UINT32_MAX) {
```

Comment:
This will always be true after 2**32 seconds. The check was meant to verify that the relative time did not overflow, but does not make sense now.

Reply:
Isnt the parameter to `pm_policy_event_update` in cycles anyway so does checking if µs overflow make sense?

Reply:
Yes, this check does not make sense now. The old API used relative time in microseconds, hence the check to see if we needed to reschedule registering the event closer to the event time.

Reply:
@knutel-nordic is it ok to shift this to a follow up PR? missed that with the MPSL changes now in nrfxlib those UT will fail on main, and are blocking the upmerge. ( merging this PR would fix it)

Reply:
@KyraLengfeld For the followup PR:

Note that `CONFIG_NRF_GRTC_TIMER` also selects `TIMER_HAS_64BIT_CYCLE_COUNTER`, `SYS_CLOCK_HW_CYCLES_PER_SEC` will be `1000000` and `SYS_CLOCK_TICKS_PER_SEC` will be `10000`. `CONFIG_TIMEOUT_64BIT` is probably also "y".

This means that in this particular case, 1 cycle is 1 us. The time parameter to `pm_policy_event_register` etc is 32 bits and will therefore overflow if the event is too far into the future (> ~4294 seconds), so we still need to reschedule ourselves closer to the event time and try again.

`CONFIG_TIMEOUT_64BIT=y` will affect the bit width of the timeout given to `mpsl_work_schedule`, which will in this case be 64 bits, too. I think we can schedule the retry only once, since `(event_time_abs_us - now_us)` converted to ticks will not overflow.